src/clib/pio_spmd.c

```diff
@@ -198,99 +215,125 @@ int pio_fc_gatherv( void *sendbuf, const int sendcnt, const MPI_Datatype sendtyp
 }
 
-///
-///  @brief Returns the smallest power of 2 greater than i
-///
+/**
+ * Returns the smallest power of 2 greater than i.
+ */
 int ceil2(const int i)
 {
-    int p=1;
-    while(p<i){
-        p*=2;
-    }
+    int p = 1;
+    while (p < i)
+        p *= 2;
+
     return(p);
 }
 
-///
-///  @brief Given integers p and k between 0 and np-1,
-///  if (p+1)^k <= np-1 then return (p+1)^k else -1
+/**
+ *  @brief Given integers p and k between 0 and np-1,
+ *  if (p+1)^k <= np-1 then return (p+1)^k else -1
+ */
 int pair(const int np, const int p, const int k)
 {
-    int q = (p+1) ^ k ;
-    int pair = (q > np-1)? -1: q;
+    int q = (p + 1) ^ k ;
+    int pair = (q > np - 1) ? -1 : q;
     return pair;
 }
 
 /**
- **  @brief Provides the functionality of MPI_Alltoallw with flow control options
+ * Provides the functionality of MPI_Alltoallw with flow control
+ * options. Generalized all-to-all communication allowing different
+ * datatypes, counts, and dis‐ placements for each partner
+ *
+ * @param sndbuf starting address of send buffer
+ * @param sndlths integer array equal to the group size specifying the
+ * number of elements to send to each processor
+ * @param sdispls integer array (of length group size). Entry j
+ * specifies the displacement in bytes (relative to sendbuf) from
+ * which to take the outgoing data destined for process j
+ * @param stypes array of datatypes (of length group size). Entry j
+ * specifies the type of data to send to process j
+ * @param rcvbuf address of receive buffer
+ * @param rcvlths integer array equal to the group size specifying the
+ * number of elements that can be received from each processor
+ * @param rdispls integer array (of length group size). Entry i
+ * specifies the displacement in bytes (relative to recvbuf) at which
+ * to place the incoming data from process i
+ * @param rtypes array of datatypes (of length group size). Entry i
+ * specifies the type of data received from process i
+ * @param comm MPI communicator for the MPI_Alltoallw call.
+ * @param handshake
+ * @param isend
+ * @param max_requests
+ * @returns 0 for success, error code otherwise.
  */
-int pio_swapm(void *sndbuf,   int sndlths[], int sdispls[],  MPI_Datatype stypes[],
-              void *rcvbuf,  int rcvlths[],  int rdispls[],  MPI_Datatype rtypes[],
-              MPI_Comm comm,const  bool handshake, bool isend,const  int max_requests)
+int pio_swapm(void *sndbuf, int *sndlths, int *sdispls, MPI_Datatype *stypes,
+              void *rcvbuf, int *rcvlths, int *rdispls, MPI_Datatype *rtypes,
+              MPI_Comm comm, const bool handshake, bool isend, const int max_requests)
 {
-
     int nprocs;
     int mytask;
-
-    int maxsend=0;
-    int maxrecv=0;
+    int maxsend = 0;
+    int maxrecv = 0;
 
     CheckMPIReturn(MPI_Comm_size(comm, &nprocs),__FILE__,__LINE__);
     CheckMPIReturn(MPI_Comm_rank(comm, &mytask),__FILE__,__LINE__);
 
-
-    if(max_requests == 0) {
+    if (max_requests == 0)
```

Comment:
if max_requests == 0 no throttling is requested and the default mpi_alltoallw function is used.